atorch/examples/llama2/README.md

```diff
@@ -423,4 +423,104 @@ sh ds_3d_llama2_entry.sh
+    bayes_opt_sg_llama2.py \
+    --dataset_path $DATASET_DIR \
+    --config_name $PRETRAINED_MODEL_DIR \
+    --tokenizer_name $PRETRAINED_MODEL_DIR \
+    --num_train_epochs 3 \
+    --block_size 2048 \
+    --total_train_batch_size $TOTAL_TRAIN_BATCH_SIZE \
+    --per_device_eval_batch_size 4 \
+    --seed 42 \
+    --preprocessing_num_workers 12 \
+    --dataloader_num_workers 0 \
+    --output_dir /tmp/test-llama2 \
+    --ignore_mismatched_sizes \
+    --ignore_dryrun_on_load_strategy
+
+```
+Note that PRETRAINED_MODEL_DIR and DATASET_DIR are the paths of model and dataset respectively. Also BO_SG_MAX_IETR is the maximum search rounds of the Bayesian optimization and RANDOM_SAMPLE is the initial sampling steps of BO. The block size is the sequence length of the input data. In the `llama2_clm.py`, the auto_accelerate function is called as follows
+
+```python
+    status, result, best_strategy = auto_accelerate(
+        model,
+        torch.optim.AdamW,
+        train_dataset,
+        loss_func=my_loss_func,
+        prepare_input=my_prepare_input,
+        model_input_format="unpack_dict",
+        optim_args={"lr": args.learning_rate},
+        optim_param_func=partial(optim_param_func, args=args),
+        dataloader_args=dataloader_args,
+        excluded=[],
+        included=[{"atorch_wrap_cls": ('LlamaDecoderLayer')}],
+        verbose=True
+    )
+```
+where `strategy` is set None, and included/excluded is used to specify the strategies to be searched. Specifically, `excluded=['zero1']` means zero1 is not in the candidates and never be chosen in the search. `included=['zero1']` means that zero1 is chosen in zero optimization method group.
+
+
+### Scripts
+
+- training file [bayes_opt_sg_llama2.py](bayes_opt_sg_llama2.py)
+
+- launch script [bayes_opt_sg_llama2_entry.sh](bayes_opt_sg_llama2_entry.sh)
+
+
+### Results
+With the above scripts, the best strategy is found as follows
+
+| batch size per gpu               | Strategy |  DryRun Throughput (samples/s)  | 
+|:----------------------------|:-----------------:|:-------:|
+| 4  |     module_replace+ amp_native+zero2+ddp   | 11.28  |
+
+
+Strategy:
+```
+        amp        zero             parallel_mode  module_replace  checkpoint
+0   NotChosen   NotChosen  {"data": 8, "tensor": 1}       NotChosen   NotChosen
+1  amp_native  zero2_fsdp  {"data": 8, "tensor": 1}  module_replace  checkpoint
+2  amp_native        fsdp  {"data": 8, "tensor": 1}  module_replace  checkpoint
+3  amp_native  zero2_fsdp  {"data": 8, "tensor": 1}  module_replace  checkpoint
+4  amp_native        fsdp  {"data": 8, "tensor": 1}  module_replace  checkpoint
+5  amp_native   NotChosen  {"data": 8, "tensor": 1}  module_replace   NotChosen
+6  amp_native       zero1  {"data": 8, "tensor": 1}  module_replace  checkpoint
+7  NotChosen        zero2  {"data": 8, "tensor": 1}  NotChosen         NotChosen
+```
+
+Corresponding Dryrun Throughput
+
+```
+[[ -0.        ]
+ [-11.28738554]
```

Comment:
Why throughput is negative?

Reply:
The BO optimizer is trying to minimize the object. Maxmizing equals miniming minus.

It is indeed confusing. I will correct it in the readme.